### PR TITLE
putty: Update to 0.78

### PR DESCRIPTION
**Pull request by svraka**
This PR updates PuTTY to the latest release, with some other necessary changes, as the project switched to CMake. I did not port the existing patch to the new build system, since Windows 7 is no longer supported in MSYS2.

In recent versions [PuTTY introduced](https://www.chiark.greenend.org.uk/~sgtatham/putty/wishlist/win-command-prompt.html) `pterm.exe`, "a PuTTY-like wrapper program for Windows command prompts" based on the Windows ConPTY. I have not looking into how we could build this on MINGW, currently it is skipped (`ConPTY not available; cannot build Windows pterm`). I'm happy to update the PR if anyone can point me to resources on this.

### CODE REVIEW

File: mingw-w64-putty/PKGBUILD [hunk 1]
```diff
@@ -6,42 +6,37 @@
...
   cp -r "${srcdir}"/${_realname}-${pkgver}/* "${srcdir}"/${_realname}-build-${MSYSTEM}
 }
 
 build() {
   cd "${srcdir}"/${_realname}-build-${MSYSTEM}
 
-  ./mkfiles.pl
-  cd windows
-  make -f Makefile.mgw CC="${CC}"
+  ${MINGW_PREFIX}/bin/cmake.exe .
```
Comment (MehdiChinoune) [inline mingw-w64-putty/PKGBUILD:33]:
```suggestion
  declare -a _extra_config
  if check_option "debug" "n"; then
    _extra_config+=("-DCMAKE_BUILD_TYPE=Release")
  else
    _extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
  fi

  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
  ${MINGW_PREFIX}/bin/cmake \
    -GNinja \
    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
    ${_extra_config[@]} \
    ../${_realname}-${pkgver}
```

Comment (svraka) [reply mingw-w64-putty/PKGBUILD]: Thanks, applied the patch.

File: mingw-w64-putty/PKGBUILD [hunk 2]
```diff
@@ -6,42 +6,37 @@
...
+  ${MINGW_PREFIX}/bin/cmake.exe --build .
 }
 
 package() {
-  cd "${srcdir}"/${_realname}-build-${MSYSTEM}/windows
-  install -D -m644 ../LICENCE "${pkgdir}${MINGW_PREFIX}"/share/licenses/${_realname}/LICENSE
+  cd "${srcdir}"/${_realname}-build-${MSYSTEM}
+  install -D -m644 LICENCE "${pkgdir}${MINGW_PREFIX}"/share/licenses/${_realname}/LICENSE
   for f in putty puttygen pageant plink pscp psftp puttytel
```
[the remark this answers is not in the document]
Comment (svraka) [reply mingw-w64-putty/PKGBUILD:53]: I'm not familiar with CMake but it seems to me PuTTY doesn't have a proper install target on Windows. They define a function [`installed_program()`](https://git.tartarus.org/?p=simon/putty.git;a=blob;f=cmake/platforms/windows.cmake;h=a7ed7c7c11ba77be9664b410c5ef9c73979eaf6b;hb=refs/tags/0.78#l195), which is called in [the Windows-specific `CMakeLists.txt`](https://git.tartarus.org/?p=simon/putty.git;a=blob;f=windows/CMakeLists.txt;h=d34b4106c03b7c9b8b8f7f404d48910d79b46899;hb=refs/tags/0.78) and it simply writes the names of the binaries in a `shipped.txt` file. We could read its contents to automatically populate the list of installed files but I don't know how the loop could be avoided.

**Comment by svraka** [comment]
Thanks for the review!

**Pull request merged**